pymchelper/readers/topas.py

```diff
@@ -0,0 +1,172 @@
+import os
+import re
+
+import numpy as np
+from pymchelper.axis import MeshAxis
+from pymchelper.page import Page
+from pymchelper.readers.common import Reader, ReaderFactory
+
+
+class TopasReaderFactory(ReaderFactory):
+    """Factory for TopasReader"""
+
+    def get_reader(self):
+        """Return TopasReader if the file extension is .csv"""
+        if self.filename.endswith('.csv'):
+            return TopasReader
+        return None
+
+
+class TopasReader(Reader):
+    """Reader for Topas output files"""
+
+    def __init__(self, filename):
+        super(TopasReader, self).__init__(filename)
+        self.directory = os.path.dirname(filename)
+
+    def get_bins(self, dimensions, bins_data, output_data):  # skipcq: PYL-R0201
+        """Get number of bins, bin size and unit for each dimension"""
+        for dimension in dimensions:
+            pattern = f"# {dimension} in (\\d+) bin[s ] of ([\\d.]+) (\\w+)"
```

Comment:
```suggestion
        pattern = f"# {dimension} in (\\d+) bin[s ] of ([\\d.]+) (\\w+)"
        for dimension in dimensions:           
```

no need to redefine pattern for every pass of the loop